flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -214,7 +216,19 @@ private RestClusterClient(
         this.waitStrategy = checkNotNull(waitStrategy);
         this.clusterId = checkNotNull(clusterId);
 
-        this.clientHAServices = checkNotNull(clientHAServices);
+        this.clientHAServices =
+                clientHAServicesFactory.create(
+                        configuration,
+                        new FatalErrorHandler() {
+                            @Override
+                            public void onFatalError(Throwable exception) {
+                                webMonitorLeaderRetriever.handleError(
+                                        new FlinkException(
+                                                "Fatal error happened with client HA "
+                                                        + "services.",
+                                                exception));
+                            }
```

Comment:
I'd still use a lambda for a bit less boilerplate here.

Reply:
Get your meaning , thanks